lib/iris/fileformats/netcdf.py

```diff
@@ -1793,78 +1824,136 @@ def _create_cf_data_variable(self, cube, dimension_names, local_keys=None,
         while cf_name in self._dataset.variables:
             cf_name = self._increment_name(cf_name)
 
+        if 'pack_dtype' in kwargs:
+            datatype = kwargs.pop('pack_dtype')
+        else:
+            datatype = None
+
+        def set_local_attrs(cf_var, local_keys):
+            """ Attributes add_offset and/or scale_factor must be set
+                after netCDF variable creation but before data is saved
+                to the netCDF Dataset.
+            """
+            if cube.standard_name:
+                cf_var.standard_name = cube.standard_name
```

Comment:
we have run into compatibility issues with Ptyhon2 Python3 and netcdf4Python 1.2 around the setting of attributes on netcdf variable which has caused some concern.
we have recently adopted a different pattern for setting attributes, to ensure that typing is preserved through the py netcdf4 testing matrix
see
#2158 #2179 #2183 
for the somewhat disjointed changes for this
(with apologies from me for the lack of cohesion in change, I missed a couple of key issues along the way)

with this in mind, please may you use the
```
_setncattr(cfvar, 'standard_name', cube.standard_name)
```
pattern for setting attributes?